serve/graph/schema.resolvers.go

```diff
@@ -0,0 +1,150 @@
+package graph
+
+// This file will be automatically regenerated based on the schema, any resolver implementations
+// will be copied through when generating and any unknown code will be moved to the end.
+// Code generated by github.com/99designs/gqlgen version v0.17.43
+
+import (
+	"context"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/gnolang/tx-indexer/serve/graph/model"
+	"github.com/gnolang/tx-indexer/types"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+)
+
+// Transactions is the resolver for the transactions field.
+func (r *queryResolver) Transactions(ctx context.Context, filter model.TransactionFilter) ([]*model.Transaction, error) {
+	it, err := r.
+		store.
+		TxIterator(
+			int64(dereferenceInt(filter.FromBlockHeight)),
+			int64(dereferenceInt(filter.ToBlockHeight)),
+			uint32(dereferenceInt(filter.FromIndex)),
+			uint32(dereferenceInt(filter.ToIndex)),
+		)
+	if err != nil {
+		return nil, gqlerror.Wrap(err)
+	}
+	defer it.Close()
+
+	fgw := dereferenceInt(filter.FromGasUsed)
+	tgw := dereferenceInt(filter.ToGasWanted)
+	fgu := dereferenceInt(filter.FromGasUsed)
+	tgu := dereferenceInt(filter.ToGasUsed)
+
+	var out []*model.Transaction
+	i := 0
+	for {
+		if i == maxElementsPerQuery {
+			graphql.AddErrorf(ctx, "max elements per query reached (%d)", maxElementsPerQuery)
+			return out, nil
+		}
+
+		if !it.Next() {
+			return out, it.Error()
+		}
+
+		select {
+		case <-ctx.Done():
+			graphql.AddError(ctx, ctx.Err())
+			return out, nil
+		default:
+			t, err := it.Value()
+			if err != nil {
+				graphql.AddError(ctx, err)
+				return out, nil
+			}
+
+			if !(t.Response.GasUsed >= int64(fgu) && (tgu == 0 || t.Response.GasUsed <= int64(tgu))) {
+				continue
+			}
+			if !(t.Response.GasWanted >= int64(fgw) && (tgw == 0 || t.Response.GasWanted <= int64(tgw))) {
+				continue
+			}
```

Comment:
Suggest setting tgu/tgw to MaxInt64 before the for loop (avoiding the check with 0 here) and turning this around, ie. 

```
			if t.Response.GasUsed >= int64(fgu) && t.Response.GasUsed <= int64(tgu) &&
				t.Response.GasWanted >= int64(fgw) && t.Response.GasWanted <= int64(tgw) {
				out = append(out, model.NewTransaction(t))
				i++
			}
```

Reply:
I tried to turn it around, but I feel that the actual version is more clear about what it is doing. Changed tgu/tgw to MaxInt if they were zero tho.